pkg/daemon/daemon.go

```diff
@@ -1905,7 +1935,7 @@ func (dn *Daemon) updateConfigAndState(state *stateAndConfigs) (bool, error) {
 	if err == nil {
 		state.currentConfig = odc.currentConfig
 		state.currentImage = odc.currentImage
-	} else {
+	} else if err != nil && !os.IsNotExist(err) {
```

Comment:
I think we should actually hard fail here, since this is during the update flow, and that file should be there. If it isn't, there's something wrong during the update itself.

Reply:
This makes sense!!

Reply:
Just a note: update writes the currentconfig to disk --> extract from the on-disk currentconfig to compare whether it is already in desired state --> if not/missing currentconfig --> trigger an update to update/rewrite to it 
Question: I recall I add this in b/c I think if the state.current = state.desired, it also means that the update is complete. Whether the currentconfig is on disk or not does not matter that much.

Reply:
I think this nill allowing needs to be there:
`I1012 15:33:12.905489   37245 daemon.go:1872] Validating against current config rendered-worker-862813a853ebf95c28c17db49406ad29
I1012 15:33:12.905521   37245 daemon.go:1785] SSH key location ("/home/core/.ssh/authorized_keys.d/ignition") up-to-date!
I1012 15:33:13.253950   37245 rpm-ostree.go:308] Running captured: rpm-ostree kargs
I1012 15:33:13.286453   37245 update.go:1977] Validated on-disk state
I1012 15:33:13.287851   37245 daemon.go:1939] Error reading config from disk
E1012 15:33:13.287870   37245 writer.go:226] Marking Degraded due to: error reading config from disk: open /etc/machine-config-daemon/currentconfig: no such file or directory
W1012 15:34:13.298089   37245 daemon.go:1674] Failed to persist NIC names: open /etc/systemd/network: no such file or directory
I1012 15:34:13.301633   37245 daemon.go:1345] Previous boot ostree-finalize-staged.service appears successful
I1012 15:34:13.301647   37245 daemon.go:1462] Current+desired config: rendered-worker-862813a853ebf95c28c17db49406ad29
I1012 15:34:13.301652   37245 daemon.go:1478] state: Degraded
I1012 15:34:13.301666   37245 update.go:1962] Running: rpm-ostree cleanup -r
Deployments unchanged.
`

Reply:
This is because the function updateConfigAndState is called at two places: checkStateOnFirstRun & performPostConfigChangeAction updateConfigAndState. When called in checkStateOnFristRun, this func allows odc = nil. When taking out, the node will degrade by the call: 
```
I1012 15:30:28.834489   37245 daemon.go:1872] Validating against current config rendered-worker-862813a853ebf95c28c17db49406ad29
I1012 15:30:28.834666   37245 daemon.go:1785] SSH key location ("/home/core/.ssh/authorized_keys.d/ignition") up-to-date!
I1012 15:30:29.170758   37245 rpm-ostree.go:308] Running captured: rpm-ostree kargs
I1012 15:30:29.202444   37245 update.go:1977] Validated on-disk state
I1012 15:30:29.203819   37245 daemon.go:1939] Error reading config from disk
E1012 15:30:29.203849   37245 writer.go:226] Marking Degraded due to: error reading config from disk: open /etc/machine-config-daemon/currentconfig: no such file or directory
I1012 15:30:31.218914   37245 daemon.go:670] Transitioned from state: Done -> Degraded
I1012 15:30:31.218944   37245 daemon.go:673] Transitioned from degraded/unreconcilable reason  -> error reading config from disk: open /etc/machine-config-daemon/currentconfig: no such file or directory
W1012 15:30:31.219031   37245 daemon.go:1674] Failed to persist NIC names: open /etc/systemd/network: no such file or directory
I1012 15:30:31.222251   37245 daemon.go:1345] Previous boot ostree-finalize-staged.service appears successful
I1012 15:30:31.222263   37245 daemon.go:1462] Current+desired config: rendered-worker-862813a853ebf95c28c17db49406ad29
I1012 15:30:31.222266   37245 daemon.go:1478] state: Degraded
I1012 15:30:31.222282   37245 update.go:1962] Running: rpm-ostree cleanup -r
Deployments unchanged.
```